great_expectations/datasource/fluent/sql_datasource.py

```diff
@@ -808,6 +808,15 @@ def qualified_name(self) -> str:
             else self.table_name
         )
 
+    @pydantic.validator("table_name", pre=True, always=True)
+    def default_table_name(cls, table_name: str, values: dict, **kwargs) -> str:
```

Comment:
```suggestion
    @pydantic.validator("table_name", pre=True, always=True)
    def _default_table_name(cls, table_name: str, values: dict, **kwargs) -> str:
```
users shouldn't ever need to call this method, so we should make it private.